Lighting/Src/lighting_controller.cpp

```diff
@@ -308,41 +504,118 @@ void HAL_TIM_PWM_PulseFinishedCallback(TIM_HandleTypeDef *htim) {
 
 void TIM6_OneSecondCallback(TIM_HandleTypeDef *htim) {
```

Comment:
Going to comment on all of the functionality you have in the 1s callback:

This is good for a demo, but it would make much much more sense to put demo code in a dedicated demo section.

We generally want to keep the timer callbacks as short as possible, they should only do what is explicitly necessary, otherwise we block the rest of the program from executing.

The core principle of the 1s callback is to synchronize the rest of the timers and provide a clear starting point for all of our other timers. htim7 in this case is going to be the timer that gets synchronized across multiple nodes, and it's one purpose should be to start other timers.